### PR TITLE
Coverage: add more exclusions

### PR DESCRIPTION
**Pull request by hugovk**
Changes proposed in this pull request:

 * With recent Coverage.py, we can use `exclude_also` instead of `exclude_lines` which requires re-enabling the standard pragma
 * `create_lut()` was a helper method not run anywhere, should we delete it instead?
 * `class TypeGuard` is compatibility code being introduced in https://github.com/python-pillow/Pillow/pull/7642. We have a `# type: ignore[no-redef]` on it, and I don't think we need to cover it in tests either.

### CODE REVIEW

File: .coveragerc [hunk 1]
```diff
@@ -2,15 +2,16 @@
...
+exclude_also =
+    # Don't complain if non-runnable code isn't run
     if 0:
     if __name__ == .__main__.:
+    # Don't complain about code that shouldn't run
+    def create_lut():
     # Don't complain about debug code
     if DEBUG:
+    # Don't complain about compatibility code
+    class TypeGuard:
```
Comment (nulano) [inline .coveragerc:14]: What do you think about this?
```suggestion
    except ImportError:
```
Most of these blocks are missing coverage in the tests already, and I'm not sure how much value there would be in adding a CI job with no optional dependencies.

Comment (nulano) [inline .coveragerc:14]: What do you think about this?
```suggestion
    # Don't complain about missing optional dependencies
    except ImportError:
```
Most of these blocks are missing coverage in the tests already, and I'm not sure how much value there would be in adding a CI job with no optional dependencies.

Comment (hugovk) [reply .coveragerc:14]: This should cover both, without the colon:
```suggestion
    # Don't complain about compatibility code for missing optional dependencies
    except ImportError
```

Comment (nulano) [reply .coveragerc]: Looking at https://app.codecov.io/gh/python-pillow/Pillow/pull/7666/indirect-changes, it seems to have worked.

**Comment by nulano** [comment]
> create_lut() was a helper method not run anywhere, should we delete it instead?

+1 for deleting it, it's easy to temporarily recreate if necessary.

**Pull request merged**